Objects/dictobject.c

```diff
@@ -2504,6 +2559,14 @@ PyDict_Copy(PyObject *o)
             _PyObject_GC_TRACK(split_copy);
         return (PyObject *)split_copy;
     }
+
+    if (PyDict_CheckExact(o) && mp->ma_values == NULL &&
+            mp->ma_keys->dk_refcnt == 1 &&
+            mp->ma_used == mp->ma_keys->dk_nentries)
```

Comment:
Am I missing some context, or is this refusing to use the fast path for shared-key dicts, when the fast path would still be good for 2/3 arrays, and the other just replaced by an INCREF?

Reply:
I don't want to complicate the code for speeding up shared-key dicts copy. Copying a shared dict is a very rare operation.